eth/ethconfig/config.go

```diff
@@ -224,6 +225,27 @@ func CreateConsensusEngine(stack *node.Node, chainConfig ctypes.ChainConfigurato
 			Epoch:  chainConfig.GetCliqueEpoch(),
 		}, db)
 	}
+	if chainConfig.GetConsensusEngineType().IsKeccak() {
+		// Otherwise assume proof-of-work
+		switch config.PowMode {
```

Comment:
I am thinking that it might be useful to move the logic of `config.PowMode` and `ethash.ModeFake` out of **ethash package** or maybe duplicate this logic.
Reusing the logic directly from ethash seems to be error prone in future.

It will be a **nice to have** if this refactor can be done in this PR, but I am also fine leaving this for **another PR** so as to not block merging of this one.

Based on your feedback @atoulme I will either accept the PR or wait for welcoming the refactor.

Reply:
I’d rather have another PR.